com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/EmberMfglib.java

```diff
@@ -80,10 +97,98 @@ public boolean doMfglibEnd() {
                 EzspMfglibEndResponse.class);
         protocolHandler.sendEzspTransaction(transaction);
 
-        EzspMfglibEndResponse mfgEndResponse = (EzspMfglibEndResponse) transaction.getResponse();
-        logger.debug(mfgEndResponse.toString());
-        if (mfgEndResponse.getStatus() != EmberStatus.EMBER_SUCCESS) {
-            logger.debug("Error with mfgEnd: {}", mfgEndResponse);
+        EzspMfglibEndResponse response = (EzspMfglibEndResponse) transaction.getResponse();
+        logger.debug(response.toString());
+        if (response.getStatus() != EmberStatus.EMBER_SUCCESS) {
+            logger.debug("Error with mfgEnd: {}", response);
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Starts transmitting an unmodulated tone on the currently set channel and power level. Upon successful return, the
+     * tone will be transmitting. To stop transmitting tone, application must call mfglibStopTone(), allowing it the
+     * flexibility to determine its own criteria for tone duration (time, event, etc.)
+     *
+     * @return true if the request succeeded
+     */
+    public boolean doMfglibStartTone() {
+        EzspMfglibStartToneRequest request = new EzspMfglibStartToneRequest();
+        EzspSingleResponseTransaction transaction = new EzspSingleResponseTransaction(request,
+                EzspMfglibStartToneResponse.class);
+        protocolHandler.sendEzspTransaction(transaction);
+
+        EzspMfglibStartToneResponse response = (EzspMfglibStartToneResponse) transaction.getResponse();
+        logger.debug(response.toString());
+        if (response.getStatus() != EmberStatus.EMBER_SUCCESS) {
+            logger.debug("Error with mfgStartTone: {}", response);
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Stops transmitting tone started by doMfglibStartTone().
+     *
+     * @return true if the request succeeded
+     */
+    public boolean doMfglibStopTone() {
+        EzspMfglibStopToneRequest request = new EzspMfglibStopToneRequest();
+        EzspSingleResponseTransaction transaction = new EzspSingleResponseTransaction(request,
+                EzspMfglibStopToneResponse.class);
+        protocolHandler.sendEzspTransaction(transaction);
+
+        EzspMfglibStopToneResponse response = (EzspMfglibStopToneResponse) transaction.getResponse();
+        logger.debug(response.toString());
+        if (response.getStatus() != EmberStatus.EMBER_SUCCESS) {
+            logger.debug("Error with mfgStopTone: {}", response);
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Starts transmitting an unmodulated tone on the currently set channel and power level. Upon successful return, the
```

Comment:
Is this maybe a copy-paste issue in the JavaDoc? It looks like the JavaDoc for sending a tone, while this is the method for starting a stream (same for stopping below).